rest/server_context.go

```diff
@@ -178,6 +181,18 @@ func NewServerContext(ctx context.Context, config *StartupConfig, persistentConf
 		sc.DatabaseInitManager = &DatabaseInitManager{}
 	}
 
+	if config.HeapProfileCollectionThreshold != nil {
+		sc.statsContext.heapProfileCollectionThreshold = int64(*config.HeapProfileCollectionThreshold)
```

Comment:
It feels like we should either define this as int64 or uint64 in both places, to avoid the risk of an unsafe cast when a user specifies a very large value in the config.

Reply:
I like using uint64 because that matches the values returned from gosputil or memlimit. The reason for not using it everywhere is that our stats are int64, so `GoMemStatsHeapInUse` is int64.